src/kraken/core/cli/option_sets.py

```diff
@@ -89,27 +89,55 @@ class GraphOptions:
     all: bool
 
     @staticmethod
-    def add_to_parser(parser: argparse.ArgumentParser, saveable: bool = True) -> None:
+    def add_to_parser(parser: argparse.ArgumentParser, saveable: bool = True, include_all_option: bool = True) -> None:
         group = parser.add_argument_group("graph options")
-        group.add_argument("tasks", metavar="task", nargs="*", help="one or more tasks to execute")
-        group.add_argument("-a", "--all", action="store_true", help="select all tasks")
         group.add_argument("--resume", action="store_true", help="load previous build state")
         group.add_argument(
             "--restart",
             choices=("all",),
             help="load previous build state, but discard existing results (requires --resume)",
         )
+
+        if include_all_option:
+            # The user can either specify a list of tasks, xor --all
+            selection = group.add_mutually_exclusive_group()
+            selection.add_argument(
+                "tasks",
+                metavar="task",
+                nargs="*",
+                help="one or more tasks to select. If not set, default tasks are selected",
+                default="",
+            )
+            selection.add_argument("--all", action="store_true", help="select all tasks")
+        else:
+            # --all may make no sense
+            # e.g. we don't want `kraken run --all` to be possible.
+            # See https://github.com/kraken-build/kraken-core/pull/37#discussion_r1171300400
+            group.add_argument(
+                "tasks",
+                metavar="task",
+                nargs="*",
+                help="one or more tasks to select. If not set, default tasks are selected",
+                default="",
+            )
+
         if saveable:
             group.add_argument("--no-save", action="store_true", help="do not save the new build state")
 
     @classmethod
     def collect(cls, args: argparse.Namespace) -> GraphOptions:
+        try:
+            all = args.all
```

Comment:
```suggestion
            all_ = getattr(args, "all", False)
```

nit: slimmer this way, and avoid overriding built ins (can pass this one liner as an argument again)